### ISSUE TITLE
Issues with key functionalities after switching identities (specific use case but very problematic)

### ISSUE THREAD

**Issue by sciencetor2** [keybase/keybase-issues#2860]
TL;DR: when switching between 2 accounts both authenticated using device keys created under paper keys, if I switch away from an account with no hosted private key, then switch back, I cannot perform cryptographic functionality like key revocations anymore.

Full Use Case:
I have 2 keybase accounts (and corresponding key families) that i use frequently. 1 of these is tied to a semi-anonymous online "persona" (yes yes i know there's no such thing as true anonymity with these things but googling it wont pop up with my house) and 1 which is tied to my personal identity that i use for "Official" communications and projects. My "Real" account uses Keybase private key hosting, my "persona" doesn't. We will refer to these as "persona" and "real" from now on. I recently had the laptop containing the original copies of the main keypairs stolen, so I got a brand new laptop, then went and logged in to persona, then logged out and logged into real, both using paper keys i kept in my lockbox for just such an occasion, and created a new device key on both. I then went to revoke the stolen keys. on real i had no problem, I was able to revoke the old key and generate the new one.  but when I switched back to persona, the function failed with an error saying i didnt have a key to perform this operation with. i tried logging out and logging in again with still no success. I ended up needing to log into ANOTHER new computer, enter paper key, create device key, then use device key to revoke my public key. this was a whole lot of extra effort.

**Comment by maxtaco** [comment]
thanks for the bug report. We hit a similar issue internally recently, and we think we've fixed it.

Cc: @patrickxb  who knows more.

**Comment by sciencetor2** [comment]
any ETA on a release?

**Comment by patrickxb** [comment]
@sciencetor2 the fix is in review and will be in the next public release.  

Sorry for the inconvenience:  it's a situation we hadn't seen at all until last week despite many tests and lots of user switching internally.

**Comment by sciencetor2** [comment]
out of curiosity, what was causing the bug?
